src/main/java/ca/nexapp/core/persistence/unitofwork/UnitOfWork.java

```diff
@@ -0,0 +1,38 @@
+package ca.nexapp.core.persistence.unitofwork;
+
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import ca.nexapp.core.persistence.unitofwork.exceptions.AlreadyInTransactionException;
+import ca.nexapp.core.persistence.unitofwork.exceptions.NotInTransactionException;
```

Comment:
The import should not present in this interface's file, and I just noticed you created a pakcage `exception` which is useless for now.